lib/rules/region.json

```diff
@@ -1,6 +1,7 @@
 {
   "id": "region",
   "selector": "body *",
+  "matches": "is-initiator-matches",
```

Comment:
This is also going to cause valid region issues to be ignored. The trick to figure out is how to distinguish these three examples:

```html
<!-- failing iframe -->
<iframe srcdoc="<h1>hello world</h1>"></iframe>

<!-- passing iframe 1 -->
<main>
  <iframe srcdoc="<h1>hello world</h1>"></iframe>
</main>

<!-- passing iframe 2 -->
  <iframe srcdoc="<main>
    <h1>hello world</h1>
</main>"></iframe>
```

Axe-core currently fails passing iframe 1, your PR fixes that, but it does so by also passing the failed iframe. I think what we'd need to do is to pass information about which frames are in a region to a new after method, and then in the after method to pass all the content of iframes that are in a region. That'll need to be recursive too.

Reply:
For this case: 
```
<!-- inapplicable comment -->
<iframe srcdoc="<-- empty frame -->"></iframe>
``` 

We can't mark this as inapplicable because we don't know the iframe is empty until after the rule has already run.